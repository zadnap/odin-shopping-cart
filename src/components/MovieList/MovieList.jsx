import MovieCard from '../MovieCard/MovieCard';
import styles from './MovieList.module.scss';

function MovieList({ movies }) {
  return (
    <section className={styles.movieList}>
      <ul className={styles.listGrid}>
        {movies.map((movie) => (
          <li key={movie.id} className={styles.movieItem}>
            <MovieCard {...movie} />
          </li>
        ))}
      </ul>
    </section>
  );
}

export default MovieList;
