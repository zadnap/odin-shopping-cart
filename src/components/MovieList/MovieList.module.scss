@use '@/styles/variables.scss' as *;

.movieList {
  display: flex;
  flex-direction: column;
  gap: var(--gap);
  flex: 1;
}
