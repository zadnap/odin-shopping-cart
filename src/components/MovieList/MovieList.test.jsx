import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import MovieList from './MovieList';

describe('MovieList component', () => {
  it('should render to match snapshot', () => {
    const mockMovies = [
      {
        id: 1,
        posterSrc: 'flow.jpg',
        title: 'Flow',
        year: 2024,
        rating: 7.9,
      },
      {
        id: 2,
        posterSrc: 'liloandstitch.jpg',
        title: 'Lilo & Stitch',
        year: 2025,
        rating: 7.2,
      },
      {
        id: 3,
        posterSrc: 'mufasa.jpg',
        title: 'Mufasa: The Lion King',
        year: 2024,
        rating: 6.6,
      },
    ];

    const { container } = render(
      <MemoryRouter>
        <MovieList movies={mockMovies} />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
