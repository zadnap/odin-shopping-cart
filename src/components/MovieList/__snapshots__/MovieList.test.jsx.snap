// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieList component > should render to match snapshot 1`] = `
<div>
  <section
    class="_movieList_6d935e"
  >
    <div
      class="_genreFilter_6d935e"
    >
      <ul
        class="_tagList_6d935e"
      >
        <li
          class="_tag_6d935e"
        >
          <button
            class="_button_ee73f7  "
          >
            Trending
          </button>
        </li>
        <li
          class="_tag_6d935e"
        >
          <button
            class="_button_ee73f7  "
          >
            Adventure
          </button>
        </li>
        <li
          class="_tag_6d935e"
        >
          <button
            class="_button_ee73f7  "
          >
            Action
          </button>
        </li>
        <li
          class="_tag_6d935e"
        >
          <button
            class="_button_ee73f7  "
          >
            Comedy
          </button>
        </li>
        <li
          class="_tag_6d935e"
        >
          <button
            class="_button_ee73f7  "
          >
            Crime
          </button>
        </li>
        <li
          class="_tag_6d935e"
        >
          <button
            class="_button_ee73f7  "
          >
            Drama
          </button>
        </li>
        <li
          class="_tag_6d935e"
        >
          <button
            class="_button_ee73f7  "
          >
            Fantasy
          </button>
        </li>
        <li
          class="_tag_6d935e"
        >
          <button
            class="_button_ee73f7  "
          >
            Fantasy
          </button>
        </li>
        <li
          class="_tag_6d935e"
        >
          <button
            class="_button_ee73f7  "
          >
            Fantasy
          </button>
        </li>
        <li
          class="_tag_6d935e"
        >
          <button
            class="_button_ee73f7  "
          >
            Fantasy
          </button>
        </li>
        <li
          class="_tag_6d935e"
        >
          <button
            class="_button_ee73f7  "
          >
            Fantasy
          </button>
        </li>
        <li
          class="_tag_6d935e"
        >
          <button
            class="_button_ee73f7  "
          >
            Fantasy
          </button>
        </li>
        <li
          class="_tag_6d935e"
        >
          <button
            class="_button_ee73f7  "
          >
            Fantasy
          </button>
        </li>
      </ul>
      <div
        class="_navigator_6d935e"
      >
        <button
          class="_button_ee73f7  "
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-left "
            data-icon="chevron-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256 246.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="_button_ee73f7  "
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-right "
            data-icon="chevron-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
    <ul
      class="_listGrid_6d935e"
    >
      <li
        class="_movieItem_6d935e"
      >
        <a
          class="_movieCard_a48f4b"
          data-discover="true"
          href="/movies/1"
        >
          <img
            alt="Flow's poster"
            class="_poster_a48f4b"
            src="flow.jpg"
          />
          <div
            class="_movieInfo_a48f4b"
          >
            <h2
              class="_title_a48f4b"
            >
              Flow
            </h2>
            <div
              class="_movieMeta_a48f4b"
            >
              <p
                class="_year_a48f4b"
              >
                2024
                 ·
              </p>
              <p
                class="_rating_a48f4b"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-star "
                  data-icon="star"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
                    fill="currentColor"
                  />
                </svg>
                 
                7.9
              </p>
            </div>
          </div>
        </a>
      </li>
      <li
        class="_movieItem_6d935e"
      >
        <a
          class="_movieCard_a48f4b"
          data-discover="true"
          href="/movies/2"
        >
          <img
            alt="Lilo & Stitch's poster"
            class="_poster_a48f4b"
            src="liloandstitch.jpg"
          />
          <div
            class="_movieInfo_a48f4b"
          >
            <h2
              class="_title_a48f4b"
            >
              Lilo & Stitch
            </h2>
            <div
              class="_movieMeta_a48f4b"
            >
              <p
                class="_year_a48f4b"
              >
                2025
                 ·
              </p>
              <p
                class="_rating_a48f4b"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-star "
                  data-icon="star"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
                    fill="currentColor"
                  />
                </svg>
                 
                7.2
              </p>
            </div>
          </div>
        </a>
      </li>
      <li
        class="_movieItem_6d935e"
      >
        <a
          class="_movieCard_a48f4b"
          data-discover="true"
          href="/movies/3"
        >
          <img
            alt="Mufasa: The Lion King's poster"
            class="_poster_a48f4b"
            src="mufasa.jpg"
          />
          <div
            class="_movieInfo_a48f4b"
          >
            <h2
              class="_title_a48f4b"
            >
              Mufasa: The Lion King
            </h2>
            <div
              class="_movieMeta_a48f4b"
            >
              <p
                class="_year_a48f4b"
              >
                2024
                 ·
              </p>
              <p
                class="_rating_a48f4b"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-star "
                  data-icon="star"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
                    fill="currentColor"
                  />
                </svg>
                 
                6.6
              </p>
            </div>
          </div>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
